server/channels/app/channel_test.go

```diff
@@ -1162,22 +1215,29 @@ func TestGetChannelMembersTimezones(t *testing.T) {
 	user := th.BasicUser
 	user.Timezone["useAutomaticTimezone"] = "false"
 	user.Timezone["manualTimezone"] = "XOXO/BLABLA"
-	th.App.UpdateUser(th.Context, user, false)
+	_, appErr := th.App.UpdateUser(th.Context, user, false)
+	require.Nil(t, appErr)
 
 	user2 := th.BasicUser2
 	user2.Timezone["automaticTimezone"] = "NoWhere/Island"
-	th.App.UpdateUser(th.Context, user2, false)
+	_, appErr = th.App.UpdateUser(th.Context, user2, false)
+	require.Nil(t, appErr)
 
 	user3 := model.User{Email: strings.ToLower(model.NewId()) + "success+test@example.com", Nickname: "Darth Vader", Username: "vader" + model.NewId(), Password: "passwd1", AuthService: ""}
-	ruser, _ := th.App.CreateUser(th.Context, &user3)
-	th.App.AddUserToChannel(th.Context, ruser, th.BasicChannel, false)
+	ruser, appErr := th.App.CreateUser(th.Context, &user3)
+	require.Nil(t, appErr)
+
+	_, appErr = th.App.AddUserToChannel(th.Context, ruser, th.BasicChannel, false)
+	require.Nil(t, appErr)
```

Comment:
This errors:
```
    channel_test.go:1231: 
        	Error Trace:	/mattermost/server/channels/app/channel_test.go:1231
        	Error:      	Expected nil, but got: &model.AppError{Id:"app.team.get_member.missing.app_error", Message:"No team member found for that user ID and team ID.", DetailedError:"", RequestId:"", StatusCode:404, Where:"AddUserToChannel", SkipTranslation:false, params:map[string]interface {}(nil), wrapped:(*store.ErrNotFound)(0xc003a0bbf0)}
        	Test:       	TestGetChannelMembersTimezones
```

Which, I think, is correct. The user needs to be added to the team first. Could you please update the code?

Reply:
I changed the check previously to expect the error, now adjusted the logic to add the user to the team first.